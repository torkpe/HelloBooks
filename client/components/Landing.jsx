import React, { Component } from 'react';
import { Link } from 'react-router';

const mdlStyle = `
  mdl-button
  mdl-js-button
  mdl-button--raised
  mdl-button--colored
`;
/**
 * returns the Landing page
 * @param {object} props
 * @return {XML} jsx
 */
const Landing = props => (
  <div>
    <div className="page-content body overlay">
      <div className="mdl-grid">
        <div className="content landing-title">
          <h1> HelloBooks</h1>
          <hr />
          <h3 id="green">...read a book today</h3>
          <Link to="/signin">
            <button
            name="getStarted"
            className={mdlStyle}>
                Get Started
            </button>
          </Link>
        </div>
      </div>
    </div>
  </div>
);
export default Landing;
