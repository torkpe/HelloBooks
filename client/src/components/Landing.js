import React, { Component } from 'react';
import { Link } from 'react-router';

class Landing extends Component {
    render(){
        return(
            <div className='page-content body'>
                <div className='mdl-grid'>
                        <div className='content'>
                            <h1> Hello-Books</h1>
                            <hr />
                            <h3 id='green'>...read a book today</h3>
                            <Link to = '/signin'><button  className="mdl-button mdl-js-button mdl-button--raised mdl-button--colored">
                                Get Started
                            </button></Link>
                    </div>                        
                </div>
            </div>            
        )
    }
}
export default Landing;
