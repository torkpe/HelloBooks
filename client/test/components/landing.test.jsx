import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';

import storeConfiguration from '../../store/store';
import Landing from '../../components/Landing.jsx';

configure({ adapter: new Adapter() });

describe('<Landing />', () => {
  it('renders <Landing /> component', () => {
    const wrapper = shallow(<Landing />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});