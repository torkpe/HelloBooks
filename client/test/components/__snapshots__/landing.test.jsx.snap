// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> renders <Landing /> component 1`] = `
<div>
  <div
    className="page-content body overlay"
  >
    <div
      className="mdl-grid"
    >
      <div
        className="content landing-title"
      >
        <h1>
           HelloBooks
        </h1>
        <hr />
        <h3
          id="green"
        >
          ...read a book today
        </h3>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/signin"
        >
          <button
            className="
  mdl-button
  mdl-js-button
  mdl-button--raised
  mdl-button--colored
"
            id="getStarted"
            name="getStarted"
          >
            Get Started
          </button>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
